src/Stream/Topic.h

```diff
@@ -25,7 +26,15 @@ class Topic {
   Topic(Kafka::BrokerSettings const &Settings, std::string const &Topic,
         SrcToDst Map, MessageWriter *Writer, Metrics::Registrar &RegisterMetric,
         time_point StartTime, duration StartTimeLeeway, time_point StopTime,
-        duration StopTimeLeeway);
+        duration StopTimeLeeway,
+        std::unique_ptr<Kafka::ConsumerFactoryInterface> CreateConsumers)
```

Comment:
```suggestion
        std::unique_ptr<Kafka::ConsumerFactoryInterface> CreateConsumers = {})
```
What was it about changing interfaces?

Reply:
Oops, forgot about that, sorry.
Your suggestion looks a bit risky though? Isn't it better that your branch fails to build so that it is clear you need to update it to pass the dependency? (Or have I misunderstood? I haven't finished my first coffee of the day yet)

Reply:
Oops, I got the types confused. Give me a moment.